app/src/main/java/org/schabi/newpipe/fragments/list/search/filter/BaseSearchFilterUiDialogGenerator.java

```diff
@@ -0,0 +1,78 @@
+// Created by evermind-zz 2022, licensed GNU GPL version 3 or later
+
+package org.schabi.newpipe.fragments.list.search.filter;
+
+import android.content.Context;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import org.schabi.newpipe.extractor.linkhandler.SearchQueryHandlerFactory;
+import org.schabi.newpipe.extractor.search.filter.FilterGroup;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static android.util.TypedValue.COMPLEX_UNIT_DIP;
+
+public abstract class BaseSearchFilterUiDialogGenerator extends BaseSearchFilterUiGenerator {
+    private static final float FONT_SIZE_TITLE_ITEMS_IN_DIP = 20f;
+
+    protected final Map<View, View.OnClickListener> viewListeners = new HashMap<>();
+
+    protected BaseSearchFilterUiDialogGenerator(
+            final SearchQueryHandlerFactory linkHandlerFactory,
+            final Callback callback,
+            final Context context) {
+        super(linkHandlerFactory, callback, context);
+    }
+
+    protected abstract void createTitle(String name, List<View> titleViewElements);
+
+    protected abstract void createFilterGroup(FilterGroup filterGroup,
+                                              UiWrapperMapDelegate wrapperDelegate,
+                                              UiSelectorDelegate selectorDelegate);
+
+    @Override
+    protected ICreateUiForFiltersWorker createContentFilterWorker() {
+        return new BaseCreateSearchFilterUI.CreateContentFilterUI(this, context);
+    }
+
+    @Override
+    protected ICreateUiForFiltersWorker createSortFilterWorker() {
+        return new BaseCreateSearchFilterUI.CreateSortFilterUI(this, context);
+    }
+
+    @Override
+    public void onResume() {
+        for (final Map.Entry<View, View.OnClickListener> view : viewListeners.entrySet()) {
+            view.getKey().setOnClickListener(view.getValue());
+        }
+    }
+
+    @Override
+    public void onPause() {
+        for (final Map.Entry<View, View.OnClickListener> view : viewListeners.entrySet()) {
+            view.getKey().setOnClickListener(null);
+        }
+    }
```

Comment:
Why is setting/unsetting the view listeners needed here?

Reply:
> Why is setting/unsetting the view listeners needed here?

I think this was a left over. Once I created the TextView's for the Spinners I kept those TextView's for reuse. There I got a null pointer exception. Removing/ and readding solved the problem. But later I figured out I missued the Spinners TextView's  instead of using the convertView parameter of the Adapters getView() method.

So this is a left over. I will remove the listener stuff there. Thanks!